src/lib/Libdb/pgsql/db_attr.c

```diff
@@ -279,8 +279,8 @@ attrlist_to_dbarray_ex(char **raw_array, pbs_db_attr_list_t *attr_list, int keys
 			if (!tmp)
 				return -1;
 
-			val = (struct str_data *) ((char *) val + ((char *) tmp - (char *) array)); /* move val since array moved */
 			array = tmp;
+			val = (struct str_data *) ((char *) val + ((char *) tmp - (char *) array)); /* move val since array moved */
```

Comment:
I don't understand what is the point of doing `+ ((char *) tmp - (char *) array));` after `array = temp;` . Won't this always evaluate to 0?

Reply:
thanks for catching this one